text/0793-polymporphic-relations-without-inheritance.md

```diff
@@ -84,59 +84,149 @@ This RFC proposes to allow explicitly declaring polymorphic traits similar to [r
 
 Polymorphic relationships take the form:
 
 ```ts
-hasMany(baseType: string, options: { polymorphic: true, inverse: string | null })
-belongsTo(baseType: string, options: { polymorphic: true, inverse: string | null })
+hasMany(abstractType: string, options: { async: boolean, polymorphic: true, inverse: string | null })
+belongsTo(abstractType: string, options: { async: boolean, polymorphic: true, inverse: string | null })
 ```
 
-Concrete Relationships which fulfill polymorphic relationships take the form:
+Concrete Relationships which fulfill polymorphic relationships take the form below where the string supplied to `as` should be the same `abstractType` supplied to the polymorphic definition above.
 
 ```ts
-hasMany(recordWithPolymorphicRelationship: string, options: { inverse: string, as: string })
-belongsTo(recordWithPolymorphicRelationship: string, options: { inverse: string, as: string })
+hasMany(recordWithPolymorphicRelationship: string, options: { async: boolean, inverse: string, as: string })
+belongsTo(recordWithPolymorphicRelationship: string, options: { async: boolean, inverse: string, as: string })
 ```
 
 So for instance, given a field named "tagged" that is a polymorphic hasMany with baseType "taggable", and inverse key "tags" the following would be the polymorphic relationship and concrete relationship.
 
 ```ts
 // polymorphic relationship
 class Tag extends Model {
-   @hasMany("taggable", { polymorphic: true, inverse: "tags" }) tagged;
+   @hasMany("taggable", { async: false, polymorphic: true, inverse: "tags" }) tagged;
 }
 // an inverse concrete relationship
 class Post extends Model {
-   @hasMany("tag", { inverse: "tagged", as: "taggable" }) tags;
+   @hasMany("tag", { async: false, inverse: "tagged", as: "taggable" }) tags;
 }
 ```
 
 ### Polymorphic Relationships without Inverses
 
 - polymorphic relationships need not have an inverse, in which case they must specify `inverse: null`.
 
-### Base Types
+### Abstract Types
 
-Every polymorphic relationship has an implicit "base type" that represents the entity on the other side of the relationship.
+Every polymorphic relationship has an implicit "abstract type" that represents the entity on the other side of the relationship.
 
-In the below example, the polymporphic relationship implies the existence of the `base type` "taggable".
+In the below example, the polymporphic relationship implies the existence of the `abstract type` "taggable".
 
+```ts
+class Tag extends Model {
+  @hasMany("taggable", { async: false, inverse: "tags", polymorphic: true }) taggables;
+}
 ```
+
+It is only required that your app implement a model (or if using instantiateRecord and the schema service implement handling for that type) if your API will return entities whose type is the abstract type. If the abstract type is never directly interacted with, it is not required to implement a Model or other associated support.
```

Comment:
> If the abstract type is never directly interacted with

What does it mean?

Reply:
This is actually carry over from the original rfc (I cleaned it up to refer to abstract vs base as it was more clear)

what it means is that if Post implements Taggable but no attempt to fetch something with type taggable and no data received will have type taggable then it’s never necessary to have a Model named Taggable

Reply:
super! Thank you for clarification!